### PR TITLE
Code style change: Replace clang-format off/on with TODO/TEST comments.

### PR DESCRIPTION
**Pull request by pgScorpio**
Changes as discussed in issue #2591

**Short description of changes**
Replace clang-format off/on with TODO/TEST comments where applicable.

CHANGELOG: Internal: Code style: TODO and TEST comments instead of un-indenting with clang-format off/on

**Context: Fixes an issue?**
partly solves clang-format problems (#2587)

**Does this change need documentation? What needs to be documented and how?**
Maybe we should document this new coding-style rule somewhere?

**Status of this Pull Request**

Ready to merge
**What is missing until this pull request can be merged?**
Review

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: linux/sound.cpp
Comment (ann0see) [inline linux/sound.cpp:204]:
```suggestion
    /* try setting buffer size seems not to work! -> no audio after this operation!
     * Doesn't this give an infinite loop? The set buffer size function will call our
     * registered callback which calls "EmitReinitRequestSignal()". In that function
     * this CSound::Init() function is called...
     */
     // jack_set_buffer_size ( pJackClient, iNewPrefMonoBufferSize );
     
```

I think this would be clearer: Explanation what to do in "/*" and the commented code in "//" (at least if possible)

File: linux/sound.cpp
Comment (ann0see) [inline linux/sound.cpp:311]:
```suggestion
                // TODO: BEGIN
                /* do not call malloc in real-time callback */
                // TODO: END
```

File: src/buffer.cpp
Comment (ann0see) [inline src/buffer.cpp:658]: Yup. After some thinking this looks quite good.

File: src/channel.cpp
Comment (ann0see) [inline src/channel.cpp:64]:
```suggestion
    // TODO: BEGIN
    /* if we later do not fire vectors in the emits, we can remove this again */
```

File: src/channel.h
Comment (ann0see) [inline src/channel.h:161]: As above

File: src/client.cpp
Comment (ann0see) [inline src/client.cpp:280]: As above.

File: src/main.cpp
Comment (ann0see) [inline src/main.cpp:819]: Previously this looked really nasty. Much better now.

File: src/protocol.cpp
Comment (ann0see) [inline src/protocol.cpp:626]: Honestly, these tests should not be in the code but somewhere externally. But that's out of scope of this PR

**Review by ann0see — COMMENTED** [review]
Yeah. Looks much better now. 

Although it's a different idea to make the todo's stand out even more:

```
// // // TODO BEGIN \\ \\ \\
/* comment */
// // // TODO END  \\ \\ \\
```

might be even more visible. I think my proposal - which you implemented here is somewhat ok, but arguably "standard-ish" and not that outstanding (after looking at the diff.)

**Comment by ann0see** [comment]
>  Maybe we should document this new coding-style rule somewhere?

Yes. CONTRIBUTING.md

File: linux/sound.cpp
Comment (pgScorpio) [reply linux/sound.cpp:312]: OK, so single comments also the closing */ on the same line ?
But that is not according your [proposal](https://github.com/jamulussoftware/jamulus/issues/2591#issuecomment-1100840078) ;=))

Comment (ann0see) [reply linux/sound.cpp:312]: I know. But after seeing it in real code, it seemed very strange.

**Comment by pgScorpio** [comment]
> Although it's a different idea to make the todo's stand out even more:
> 
> ```
> // // // TODO BEGIN \\ \\ \\
> /* comment */
> // // // TODO END  \\ \\ \\
> ```
> 
> might be even more visible. I think my proposal - which you implemented here is somewhat ok, but arguably "standard-ish" and not that outstanding (after looking at the diff. Surely, this would loose the editor highlighting)

To be honest all those slashes and back-slashes make me dizzy, even nauseous if I look too long....
But of course there are MANY other possibilities, some even more 'outstanding':
```
//<<< TODO: BEGIN >>>
//>>> TODO: END <<<

/*** TODO: BEGIN ***/
/*** TODO: END ***/

/*### TODO: BEGIN ###*/   <- this one stands out really well I think
/*=== TODO: END ===*/

```

> Surely, this would loose the editor highlighting

Not necessarily. For most highlighters the keywords are 'TODO:' and 'FIXME:' anywhere in any comment, so as long you keep it uppercase and directly followed by a colon, most highlighters will still work.

**Comment by ann0see** [comment]
I like /*** TODO: BEGIN ***/ since it looks standard-ish. However

/*### TODO: BEGIN ###*/ 

stands out well too. Would you prefer 
/*### TODO: BEGIN ###*/  ?

**Comment by pgScorpio** [comment]
> Would you prefer
> /_### TODO: BEGIN ###_/ ?

Yes it stands out more, maybe we should combine them to make BEGIN stand out the most?
```
//### TODO: BEGIN ###//
//*** TODO: END ***//
```

**Comment by ann0see** [comment]
I'd prefer the consistent approach.

**Comment by pgScorpio** [comment]
> I'd prefer the consistent approach.
OK it will be
```
//### TODO: BEGIN ###//
/* comment */
code...
//### TODO: END ###//
```
then ?
And what about */ end-of-comment?
Always at the end of the (last) comment line ?

**Comment by ann0see** [comment]
Yes. If the code already uses a specific multi line comment style this should be used. If not I'd go with your present proposal

**Comment by pgScorpio** [comment]
> Yes. If the code already uses a specific multi line comment style this should be used.

Well most of them use the repeated asterisk:
```
/************
 * multi
 * line
 * comment
 ************/

```
But I think that's over-doing it here.

**Comment by ann0see** [comment]
If it's just most I think we're fine with the lighter way — as long as clang-format is happy, I think it's ok b

**Comment by pgScorpio** [comment]
> If it's just "most" not all, I think we're fine with the lighter way

Another option is not using block comments at all:
```
//### TODO: BEGIN ###//
// comment
// comment
code...
//### TODO: END ###//
```
I don't like using block comment inside the code anyway, since you can't use block comment to comment out a piece of code containing a block comment.

**Comment by ann0see** [comment]
That's also possible

File: src/settings.cpp
Comment (ann0see) [inline src/settings.cpp:883]: The diff looks a bit strange here, but I think it's ok.

**Review by ann0see — APPROVED** [review]
Great. Thanks. Looks good now. 
Can the other PR with clang format be closed then?

**Comment by ann0see** [comment]
Note to maintainers: This should be squash merged.

**Comment by pgScorpio** [comment]
> Can the other PR with clang format be closed then?

Yes. I will close it..

File: src/settings.cpp
Comment (pgScorpio) [reply src/settings.cpp:883]: Yeah, I had too look 3 times too here...
Looks a bit strange because of the splitting up of if else.

If I remember well this was exactly the spot where also clang-format got confused ;=))

**Comment by ann0see** [comment]
@hoffie @pljones could someone please review this PR?

File: windows/sound.cpp
Comment (hoffie) [inline windows/sound.cpp:319]: It looks like the `_exit(1)` call was moved to the already rather long line. Was this done on purpose, if so which?
I'd rather have expected it to stay like this:
```suggestion
                               arg(HWBufferInfo.lMinSize).arg(HWBufferInfo.lMaxSize).arg(HWBufferInfo.lPreferredSize).arg(HWBufferInfo.lGranularity) );
    _exit(1);
```

**Review by hoffie — COMMENTED** [review]
Thanks! I've got one minor nit/question -- see inline review comment.

I guess one could bike-shed about the proper style here. The syntax looks non-natural to me, but I guess one can get used to it. In the end, it doesn't matter that much and if done consistently -- which seems to be the case -- it can easily be changed if wanted.

Ideally, such changes would be ok'ed by all maintainers, but that's rather hard right now as per #2617.
As the issue and this PR have been open for some time now, I guess it's ok to merge even without more maintainer acknowledgements.

@pgScorpio I've got two general remarks:
- If a commit and/or PR fixes an issue, you can use the "Fixes #0000" syntax in order to make Github link the PR to the Issue automatically. This will also auto-close the issue on PR-merge. I've done that manually here now.
- Our CHANGELOG helper currently only understands single lines, i.e. the actual content has to stay on the same line as the keyword for it to be used. I'm therefore re-mentioning it here with slight variation.

CHANGELOG: Code style: Use TODO and TEST comments instead of un-indenting with clang-format off/on

Note to maintainers:
- [x] Should be squashed / commit message-edited
- [x] Might want to cherry-pick the add-to-authors commit from one of the other PRs here.

I can do both once the PR is fully approved.

**Comment by ann0see** [comment]
@hoffie @pgScorpio I think the last commit made clang-format fail???

**Comment by pgScorpio** [comment]
@annosee

> @hoffie @pgScorpio I think the last commit made clang-format fail???

That's the other clang-format quirk again ! A 149 character long line on windows (CR/LF) is 150 chars long for clang-format !

Nothing I can do about that....
I think the only option is to use .gitatributes to force LF only on windows too.
See [docs.github.com](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings)

**Comment by hoffie** [comment]
Sorry for the long delay to re-review. Sadly, there are conflicts by now, most likely due to the sound backend moving. Therefore, I'm not clicking "approve" although this should be fine to do once the conflicts are fixed.

**Comment by pgScorpio** [comment]
> Sorry for the long delay to re-review. 
> Sadly, there are conflicts by now, most likely due to the sound backend moving. 

Yeah, the very disadvantage of waiting to long with merging is the risk on conflicts....
But I just rebased this PR, so it should be ok now.

**Review by hoffie — COMMENTED** [review]
I haven't investigated why, but the diff of this PR looks really strange now.
On Github it appears to re-list changes from other PRs/commits and I can't tell why (e.g. .github/autobuild/android.sh).
On CLI, it looks less confusing but there are diff hunks which revert parts of another PR:

```
$ git diff upstream/master..pr2600 src/serverdlg.cpp
diff --git a/src/serverdlg.cpp b/src/serverdlg.cpp
index 968aa2f4..6aadcd28 100644
--- a/src/serverdlg.cpp
+++ b/src/serverdlg.cpp
@@ -175,16 +175,14 @@ CServerDlg::CServerDlg ( CServer* pNServP, CServerSettings* pNSetP, const bool b
     pbtRecordingDir->setWhatsThis ( "<b>" + tr ( "Main Recording Directory" ) + ":</b> " +
                                     tr ( "Click the button to open the dialog that allows the main recording directory to be selected.  "
                                          "The chosen value must exist and be writeable (allow creation of sub-directories "
-                                         "by the user %1 is running as)." )
-                                        .arg ( APP_NAME ) );
+                                         "by the user Jamulus is running as)." ) );
 
...
```

**Comment by pgScorpio** [comment]
> I haven't investigated why, but the diff of this PR looks really strange now.

No idea what happened... I just did a rebase, and there was just one conflict in a comment. all other changes are by auto-merge...
And I guess it's normal that other changes due to rebase will show up here too?

**Comment by ann0see** [comment]
No. That's not normal. I've seen this quite often in the past and it always was a hint that this could lead to a huge screw up. What were the exact commands you were using?

**Comment by ann0see** [comment]
> [did I do a git commit here?, if so that might be the problem?]

Maybe? The PR in this state is not mergable unfortunately. Can you try the rebasing again or cherry pick the commits on a new branch?

**Comment by pgScorpio** [comment]
> Can you try the rebasing again or cherry pick the commits on a new branch?

I did a reset of the last commit...
does that solve the problem ??

**Review by hoffie — APPROVED** [review]
Thanks @pgScorpio! Looks good now.
I've squashed the commits and cherry-picked your contributors list addition from another PR.

Will merge when CI is green.

**Pull request merged**
